### ISSUE TITLE
Use logger instead of printStackTrace()

### ISSUE THREAD

**Issue by AzeemJiva** [corretto/heapothesys#18]
Replace all usage of printStackTrace() with a proper logging mechanism.

**Comment by amitvc** [comment]
@AzeemJiva Can I pick up this issue please?

**Comment by AzeemJiva** [comment]
Yes!

**Comment by amitvc** [comment]
Do we want to use java.util.logging.Logger and allow users to configure log level etc or we are open to introducing something like log4j or slf4j?

**Comment by AzeemJiva** [comment]
Keeping the number of dependencies down is desirable.  @yoroto your thoughts?

**Comment by amitvc** [comment]
@AzeemJiva I agree. I can start with what is provided in jdk. We can replace it later if we need to.

**Comment by yoroto** [comment]
@AzeemJiva Yes, we don't want to pull in those big packages, or anything would significantly affect allocation rate.
